src/components/DownloadImageDialog/ImageForm.tsx

```diff
@@ -124,29 +121,41 @@ export const ImageForm = memo(function ImageForm({
 	);
 
 	const handleVariantChange = useCallback(
-		(v: typeof variant) => {
-			setVariant(v);
-			onChange('developmentMode', v === 'dev');
+		(newVariant: typeof variant) => {
+			setVariant(newVariant);
+			const newState: Partial<DownloadImageFormModel> = {
+				developmentMode: newVariant === 'dev',
+			};
+			if (version?.hasPrebuiltVariants) {
+				const rawVersionForVariant = version.rawVersions[newVariant];
+				if (rawVersionForVariant != null) {
+					newState.version = rawVersionForVariant;
+				}
+			}
+			onChange(newState);
 		},
-		[onChange],
+		[onChange, version],
 	);
 
 	const handleVersionChange = useCallback(
 		(ver: typeof version) => {
 			ver ??=
 				versionSelectionOpts.find((v) => v.isRecommended) ??
 				versionSelectionOpts[0];
+			const newState: Partial<DownloadImageFormModel> = {
+				developmentMode: variant === 'dev',
+			};
 			if (ver?.hasPrebuiltVariants) {
 				const rawVersionForVariant = ver.rawVersions[variant];
-				if (rawVersionForVariant) {
-					onChange('version', rawVersionForVariant);
-					setVersion(ver);
-				} else {
+				if (!rawVersionForVariant) {
 					handleVariantChange(variant === 'dev' ? 'prod' : 'dev');
+					return;
 				}
-				return;
+				newState.version = rawVersionForVariant;
+			} else {
+				newState.version = ver?.rawVersion;
 			}
```

Comment:
Move the `newState.version = ver?.rawVersion;` above the `if` and get rid of the `else`

Reply:
We can't do that b/c `rawVersion` only exists in the union type when `hasPrebuiltVariants: false`, so before the `if` TS doesn't allow us to access it.